File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False]*n
    
    def dfs(pc):
        visited[pc] = True
        for idx in range(n):
            if not visited[idx] and computers[pc][idx] == 1:
                dfs(idx)
    for node in range(n):
        if not visited[node]:
            dfs(node)
            answer += 1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    private boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[computers.length];
        
        for (int i = 0; i < computers.length; i++) {
            if (!visited[i]) {
                answer++;
                dfs(i, computers);
            }
        }
        return answer;
    }
    
    public void dfs(int num, int[][] computers) {
        visited[num] = true;
        
        for (int i = 0; i < computers.length; i++) {
            if (!visited[i] && computers[num][i] == 1) {
                dfs(i, computers);
            }
        }
    }
}